test/xpu_api/random/statistics_tests/philox_uniform_real_distribution_test.pass.cpp

```diff
@@ -0,0 +1,597 @@
+{
+
+    // memory allocation
+    std::vector<oneapi::dpl::internal::element_type_t<RealType>> samples(nsamples);
+
+    constexpr int num_elems = oneapi::dpl::internal::type_traits_t<RealType>::num_elems == 0
+                                  ? 1
+                                  : oneapi::dpl::internal::type_traits_t<RealType>::num_elems;
+    // generation
+    {
+        sycl::buffer<oneapi::dpl::internal::element_type_t<RealType>, 1> buffer(samples.data(), nsamples);
+
+        queue.submit([&](sycl::handler& cgh) {
+            sycl::accessor acc(buffer, cgh, sycl::write_only);
+
+            cgh.parallel_for<>(sycl::range<1>(nsamples / num_elems), [=](sycl::item<1> idx) {
+                unsigned long long offset = idx.get_linear_id() * num_elems;
+                Engine engine;
+                engine.discard(offset);
+                oneapi::dpl::uniform_real_distribution<RealType> distr(left, right);
+
+                sycl::vec<oneapi::dpl::internal::element_type_t<RealType>, num_elems> res = distr(engine);
+                res.store(idx.get_linear_id(), acc);
+            });
+        });
+        queue.wait();
+    }
+
+    // statistics check
+    int err = statistics_check(nsamples, left, right, samples);
+
+    if (err)
+    {
+        std::cout << "\tFailed" << std::endl;
+    }
+    else
+    {
+        std::cout << "\tPassed" << std::endl;
+    }
+
+    return err;
+}
+
+template <class RealType, class UIntType, typename Engine>
+int
+test_portion(sycl::queue& queue, oneapi::dpl::internal::element_type_t<RealType> left,
+             oneapi::dpl::internal::element_type_t<RealType> right, int nsamples, unsigned int part)
+{
+
+    // memory allocation
+    std::vector<oneapi::dpl::internal::element_type_t<RealType>> samples(nsamples);
+    constexpr int num_elems = oneapi::dpl::internal::type_traits_t<RealType>::num_elems == 0
+                                  ? 1
+                                  : oneapi::dpl::internal::type_traits_t<RealType>::num_elems;
+    int n_elems = (part >= num_elems) ? num_elems : part;
+
+    // generation
+    {
+        sycl::buffer<oneapi::dpl::internal::element_type_t<RealType>, 1> buffer(samples.data(), nsamples);
+
+        queue.submit([&](sycl::handler& cgh) {
+            sycl::accessor acc(buffer, cgh, sycl::write_only);
+
+            cgh.parallel_for<>(sycl::range<1>(nsamples / n_elems), [=](sycl::item<1> idx) {
+                unsigned long long offset = idx.get_linear_id() * n_elems;
+                Engine engine;
+                engine.discard(offset);
+                oneapi::dpl::uniform_real_distribution<RealType> distr(left, right);
+
+                sycl::vec<oneapi::dpl::internal::element_type_t<RealType>, num_elems> res = distr(engine, part);
+                for (int i = 0; i < n_elems; ++i)
+                    acc[offset + i] = res[i];
+            });
+        });
+        queue.wait_and_throw();
```

Comment:
inconsistent wait() and wait_and_throw() - if we want to use throw it is better to have try - catch block

Reply:
The test was fixed